packages/cli/src/linter/base-linter.ts

```diff
@@ -0,0 +1,55 @@
+import { Spectral, Document, ISpectralDiagnostic } from '@stoplight/spectral-core'
+import { Json as JsonParser, JsonParserResult } from '@stoplight/spectral-parsers'
+import { Logger } from '../logger'
+
+export abstract class BaseLinter {
+  protected readonly spectral: Spectral
+  private readonly spectralDocument: Document<unknown, JsonParserResult<unknown>>
+  private results: ISpectralDiagnostic[] = []
+
+  protected constructor(definition: unknown) {
+    const json = JSON.stringify(definition).replaceAll('"$ref":', '"_$ref":')
+    this.spectralDocument = new Document(json, JsonParser)
+    this.spectral = new Spectral()
+  }
+
+  public async lint(): Promise<void> {
+    this.results = await this.spectral.run(this.spectralDocument)
+  }
+
+  public getResults() {
+    return this.results.map((result) => ({
+      message: result.message,
+      path: this.simplifyPath(result.path),
+      severity: +result.severity,
```

Comment:
**question:** what is that 😮 ?

Reply:
This would be the [unary plus operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus). This operator converts whatever was given to it into a number. For instance, `+"5"` is equivalent to `Number("5")`. In this case, I use it to convert the strongly-typed [DiagnosticSeverity](https://github.com/stoplightio/types/blob/master/src/diagnostics.ts#L6) enum into a plain integer. This way, I avoid leaking implementation details outside of the public `getResults` method.